src/tests/tests.h

```diff
@@ -566,6 +566,10 @@ class Test {
 
       const std::optional<CodeLocation>& registration_location() const { return m_registration_location; }
 
+      void set_serialization(bool needs_serialization) { m_needs_serialization = needs_serialization; }
+
+      bool is_serialized() const { return m_needs_serialization; }
```

Comment:
When would a test need to check if it was serialized or not? Since this is set (effectively) at runtime based on which macro is invoked.